Sources/SignalRClient/TimeScheduler.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+
+class TimeScheduler {
+    private let queue = DispatchQueue(label: "com.schduler.timer")
+    private var timer: DispatchSourceTimer?
+    private var interval: TimeInterval
+    
+    init(initialInterval: TimeInterval) {
+        self.interval = initialInterval
+    }
+    
+    func start(sendAction: @escaping () async -> Void) {
+        stop()
+        timer = DispatchSource.makeTimerSource(queue: queue)
```

Comment:
Maybe we could make the `TimeScheduler` `actor` to avoid potential concurrent issue like timer leaking when calling `start` concurrently